src/MarkovChainMonteCarlo.jl

```diff
@@ -173,8 +183,15 @@ function log_likelihood(mcmc::MCMC{FT},
         diff = g - mcmc.obs_sample
         log_rho[1] = -FT(0.5) * diff' * (mcmc.obs_noise_cov \ diff)
     else
-        log_gpfidelity = -FT(0.5) * log(det(Diagonal(gvar))) # = -0.5 * sum(log.(gvar))
-        diff = g - mcmc.obs_sample
+	# det(log(Γ))
+	# Ill-posed numerically for ill-conditioned covariance matrices with det≈0
+        #log_gpfidelity = -FT(0.5) * log(det(Diagonal(gvar))) # = -0.5 * sum(log.(gvar))
+	# Well-posed numerically for ill-conditioned covariance matrices with det≈0
+	full_cov = Diagonal(gvar)
+	eigs = eigvals(full_cov)
+	log_gpfidelity = -FT(0.5) * sum(log.(eigs))
+	# Combine got log_rho
```

Comment:
In this version of MCMC, we are actually assuming that `gvar` is a vector of floats, I'm thinking maybe we should relax this to take in matrices. Perhaps the easiest way is to rewrite the current `mcmc_sample!` function to take in matrix input `gcov` and add this extra function
```Julia 
function mcmc_sample!(mcmc::MCMC{FT}, g::Vector{FT}, gvar::Vector{FT}) where {FT}
    return mcmc_sample!(mcmc, g, Diagonal(gvar))
end
```

Reply:
I've just checked that if eigvals receives a  `Diagonal` type matrix it really does just take the diagonal as the eigenvalues. and so is a nanosecond timed computation still.

Reply:
Great!

Reply:
I am not sure you have pushed this yet? I can't see it it